### DISCUSSION TITLE
What's the purpose of PHOTOPRISM_IMPORT_DEST / --dest parameter when importing?

### DISCUSSION THREAD

**Discussion by ricardopaiva** [photoprism/photoprism#3467] [Q&A] [answered]
I'm using docker-compose to setup Photoprism.
I have my photos and videos in a structured folder following the pattern:
YYYY\YYYY-MM-DD - \<Event title\>
like:
2001\2001-06-02 - Trip to London
2023\2023-01-01 - Jeremy's Birthday

Therefore I want to have these folders indexed and for that reason I'm using these volumes in my docker-compose file:

    volumes:
      - multimedia_photos:/photoprism/originals/Photos:ro
      - multimedia_videos:/photoprism/originals/Videos:ro

Multimedia_photos and Multimedia_videos are two shared folders I have on my NAS.

Now I'm trying to set up a workflow to import the photos from my mobile phone to this multimedia_photos folder.
My idea would be to use PhotoSync to move the photos from the mobile into the Photoprism import folder. Then I would run the following command (I'm going to run this in cron but I'm testing it manually for now) to import the photos:

`docker exec -ti photoprism photoprism import '/photoprism/import' --dest '/photoprism/originals/Organizar'`

Imported photos would be, by default, organised into Year/Month/Filename, which sounds reasonable to me and on top I would need to make some extra effort to split the photos by event. I might find a better way to deal with this but for now that should work for me.
So I also have these volumes in docker-compose:

    volumes:
      - multimedia_organizar:/photoprism/originals/Organizar:rw
      - multimedia_import:/photoprism/import

So I will use PhotoSync to copy the photos to the **Import** folder. Then run the **docker exec -ti photoprism photoprism import** command, so all the photos will be moved to the **Organizar** folder, splitted into Year/Month and then I will run some additional organisation scripts to split the photos by event and finally move it to the Photos folder.

But when I run the **docker exec -ti photoprism photoprism import '/photoprism/import' --dest '/photoprism/originals/Organizar'** command, the output is the following:

```
INFO[2023-06-12T22:11:58Z] config: case-insensitive file system detected 
INFO[2023-06-12T22:11:58Z] Become a member today, support our mission and enjoy our member benefits! 💎 
INFO[2023-06-12T22:11:58Z] Visit https://www.photoprism.app/membership to learn more. 
INFO[2023-06-12T22:11:58Z] migrate: running database migrations         
INFO[2023-06-12T22:11:59Z] moving media files from /photoprism/originals/Organizar/photoprism/originals/true 
INFO[2023-06-12T22:12:00Z] classify: loading nasnet                     
INFO[2023-06-12T22:12:02Z] classify: loading labels from labels.txt     
INFO[2023-06-12T22:12:02Z] import: found no .ppignore file              
INFO[2023-06-12T22:12:02Z] completed in 4.28038526s      
```

I would expect the files to be moved to folder **/photoprism/originals/Organizar/2023/06/** but it seems like the --dest parameter is not working properly.
The folder I'm getting on the output is **/photoprism/originals/true**, which is also weird because this "true" folder doesn't make any sense and seems to be coming from a boolean variable, probably.

I also tried to run this instead:
**docker exec -ti photoprism photoprism import '/photoprism/import' --dest 'originals/Organizar'**
Because according to the documentation --dest is a path relative to the Originals path, but the output seems to be the same.

If I skip the parameter, then the import works fine but the folder is created under the /photoprism/originals folder like /photoprism/originals/2023/06 but that's not what I want because it's not importing inside folder Organizar.

**docker exec -ti photoprism photoprism import '/photoprism/import'**

Here's the output:
```
INFO[2023-06-12T22:23:27Z] config: case-insensitive file system detected 
INFO[2023-06-12T22:23:27Z] Become a member today, support our mission and enjoy our member benefits! 💎 
INFO[2023-06-12T22:23:27Z] Visit https://www.photoprism.app/membership to learn more. 
INFO[2023-06-12T22:23:27Z] migrate: running database migrations         
INFO[2023-06-12T22:23:27Z] **moving media files from /photoprism/import to /photoprism/originals/Organizar** 
INFO[2023-06-12T22:23:27Z] classify: loading nasnet                     
INFO[2023-06-12T22:23:29Z] classify: loading labels from labels.txt     
INFO[2023-06-12T22:23:29Z] import: found no .ppignore file              
INFO[2023-06-12T22:23:29Z] media: 994E28F2-CE74-4010-AD9C-EA567C130793.JPG was taken at 2023-06-09 12:09:21.515 +0000 UTC (meta) 
**INFO[2023-06-12T22:23:30Z] import: moving main jpg file 994E28F2-CE74-4010-AD9C-EA567C130793.JPG to 2023/06/20230609_120921_A72C0CD8.jpg**
**INFO[2023-06-12T22:23:32Z] import: updated main jpg file 2023/06/20230609_120921_A72C0CD8.jpg** 
INFO[2023-06-12T22:24:36Z] completed in 1m9.515612866s  
```

Do you feel like this is a bug? Or am I missing the point of this --dest parameter when importing?

**Comment by lastzero** [comment, marked as answer]
**Thanks for reporting this!** An updated [development preview build](https://docs.photoprism.app/getting-started/updates/#development-preview) will be available for testing soon:

- https://docs.photoprism.app/release-notes/#development-preview

**Reply by lastzero** [reply]
We've started testing the current development build, so we'll probably be able to release an update tomorrow.

**Reply by ricardopaiva** [reply]
Hi @lastzero ,

It seems like this is not working properly yet.
So I tried to run the following commands to import the photos:
`docker exec -ti photoprism photoprism import '/photoprism/import' --dest 'Organizar'`
And also tried like this:
`docker exec -ti photoprism photoprism import '/photoprism/import' --dest './Organizar'`

I was expecting the photos to be moved to /photoprism/originals/Organizar. But that's not what's happening.
The output is the following:
```
INFO[2023-06-18T21:01:35Z] config: case-insensitive file system detected 
INFO[2023-06-18T21:01:35Z] Become a member today, support our mission and enjoy our member benefits! 💎 
INFO[2023-06-18T21:01:35Z] Visit https://www.photoprism.app/membership to learn more. 
INFO[2023-06-18T21:01:35Z] migrate: running database migrations         
INFO[2023-06-18T21:01:36Z] moving media files from /photoprism/import to /photoprism/originals/Organizar 
INFO[2023-06-18T21:01:36Z] classify: loading nasnet                     
INFO[2023-06-18T21:01:41Z] classify: loading labels from labels.txt     
INFO[2023-06-18T21:01:41Z] import: found no .ppignore file              
INFO[2023-06-18T21:01:41Z] media: 994E28F2-CE74-4010-AD9C-EA567C130793.JPG was taken at 2023-06-09 12:09:21.515 +0000 UTC (meta) 
INFO[2023-06-18T21:01:41Z] import: moving main jpg file 994E28F2-CE74-4010-AD9C-EA567C130793.JPG to 2023/06/20230609_120921_A72C0CD8.jpg 
INFO[2023-06-18T21:01:46Z] import: updated main jpg file 2023/06/20230609_120921_A72C0CD8.jpg 
INFO[2023-06-18T21:02:49Z] completed in 1m13.616422781s
```

This seems to be correct:
`INFO[2023-06-18T21:01:36Z] moving media files from /photoprism/import to /photoprism/originals/Organizar`
But the media is not being moved to this folder.
Instead of
`INFO[2023-06-18T21:01:41Z] import: moving main jpg file 994E28F2-CE74-4010-AD9C-EA567C130793.JPG to 2023/06/20230609_120921_A72C0CD8.jpg`
I was expecting to have:
`INFO[2023-06-18T21:01:41Z] import: moving main jpg file 994E28F2-CE74-4010-AD9C-EA567C130793.JPG to **Organizar/**2023/06/20230609_120921_A72C0CD8.jpg`

So if I run this command:
`docker exec -ti photoprism ls /photoprism/originals`
I get this:
`2023  Organizar  Photos  Videos`
But the 2023 folder should be created inside the "Organizar" folder, unless I'm misunderstanding the --dest flag.

**Reply by lastzero** [reply]
The flag comes first, then the folder. If you use the default folder, I think you shouldn't need it at all.

**Reply by ricardopaiva** [reply]
I'm not using the default folder.
If I skip the flag, then the default folder will be used so the photos will be imported into /photoprism/originals so I will have the following folder structure:
/photoprism/originals/2023/06/20230609_120921_A72C0CD8.jpg

But I want the photos to be imported to a subfolder and therefore I'm including the --dest flag with './Organizar'.
I expect the photos to be imported to:
/photoprism/originals/**Organizar/**2023/06/20230609_120921_A72C0CD8.jpg

But that's not happening. The subfolder "Organizar" is being ignored.

**Reply by lastzero** [reply]
I meant the default IMPORT folder.

**Reply by lastzero** [reply]
Here's a complete example based on our latest release (running in our development environment):

```
$ photoprism import --dest=Foo 
INFO[2023-06-18T22:37:26Z] migrate: running database migrations         
INFO[2023-06-18T22:37:26Z] moving media files from /go/src/github.com/photoprism/photoprism/storage/import to /go/src/github.com/photoprism/photoprism/storage/originals/Foo 
INFO[2023-06-18T22:37:26Z] classify: loading nasnet                     
INFO[2023-06-18T22:37:30Z] classify: loading labels from labels.txt     
INFO[2023-06-18T22:37:30Z] import: found no .ppignore file              
INFO[2023-06-18T22:37:30Z] import: added folder /                       
INFO[2023-06-18T22:37:30Z] import: added folder /2011                   
INFO[2023-06-18T22:37:30Z] import: added folder /2011/10                
INFO[2023-06-18T22:37:30Z] import: added folder /Google Takeout 22.01.2001 
INFO[2023-06-18T22:37:30Z] media: 20111002_120138_783F539B.jpg was taken at 2011-10-02 12:01:38.34 +0000 UTC (meta) 
INFO[2023-06-18T22:37:30Z] import: 20111002_120138_783F539B.jpg is identical to 2011/10/20111002_120138_783F539B.jpg (sha1 7941e0a5acaf92409a235074c82ebcfde76b4fdf) 
INFO[2023-06-18T22:37:30Z] import: deleted 2011/10/20111002_120138_783F539B.jpg (already exists) 
INFO[2023-06-18T22:37:30Z] media: animated-earth.gif was taken at 2022-04-15 03:03:32 +0000 UTC (file mod time) 
INFO[2023-06-18T22:37:30Z] media: IMG_6909.JPG was taken at 2016-11-20 22:37:19 +0000 UTC (meta) 
INFO[2023-06-18T22:37:30Z] import: IMG_6909.JPG is identical to 'Photos/New folder Fotos: Fügen/demo/20161120-163719-Chicago-Usa-2016.jpg' (sha1 3cfc64b483e1bfa26f0fe7a8ce73fe8053f23a45) 
INFO[2023-06-18T22:37:30Z] import: deleted 'Google Takeout 22.01.2001/IMG_6909.JPG' (already exists) 
INFO[2023-06-18T22:37:30Z] folder: added album 'April 2022' (path:Foo/2022/04 public:true) 
INFO[2023-06-18T22:37:30Z] import: created folder /Foo/2022/04          
INFO[2023-06-18T22:37:30Z] import: moving main gif file 'Google Takeout 22.01.2001/animated-earth.gif' to Foo/2022/04/20220415_030332_4C1DD8E8.gif 
INFO[2023-06-18T22:37:30Z] media: IMG_6964.JPG was taken at 2016-11-21 16:34:34 +0000 UTC (meta) 
INFO[2023-06-18T22:37:30Z] import: IMG_6964.JPG is identical to 'Photos/New folder Fotos: Fügen/demo/20161121-103434-Wacker-Chicago-2016.jpg' (sha1 2f7ca8c3abf510ee73ab99799d54d2b8480356eb) 
INFO[2023-06-18T22:37:30Z] convert: converting 20220415_030332_4C1DD8E8.gif to 20220415_030332_4C1DD8E8.gif.jpg (gif) 
INFO[2023-06-18T22:37:30Z] import: deleted 'Google Takeout 22.01.2001/IMG_6964.JPG' (already exists) 
INFO[2023-06-18T22:37:30Z] convert: 20220415_030332_4C1DD8E8.gif.jpg created in 22.652419ms (gif) 
DEBU[2023-06-18T22:37:30Z] import: created 20220415_030332_4C1DD8E8.gif.jpg 
INFO[2023-06-18T22:37:30Z] media: IMG_6810.JPG was taken at 2016-11-20 18:35:00 +0000 UTC (meta) 
INFO[2023-06-18T22:37:31Z] import: IMG_6810.JPG is identical to 'Photos/New folder Fotos: Fügen/demo/20161120-123500-Sir-Georg-Solti-Garden-Chicago-2016.jpg' (sha1 8dff97cba8d1440272b4eee90583fa134b2caf7f) 
INFO[2023-06-18T22:37:31Z] import: deleted 'Google Takeout 22.01.2001/IMG_6810.JPG' (already exists) 
INFO[2023-06-18T22:37:31Z] media: changing-of-the-guard--buckingham-palace_7925318070_o.jpg was taken at 2012-09-02 12:40:44.81 +0000 UTC (meta) 
INFO[2023-06-18T22:37:31Z] import: changing-of-the-guard--buckingham-palace_7925318070_o.jpg is identical to 2012/09/20120902_124044_AF7D1566.jpg (sha1 e35df28579c2a5f5ef1428d2eebd690eb15422a9) 
INFO[2023-06-18T22:37:31Z] import: deleted 'Google Takeout 22.01.2001/changing-of-the-guard--buckingham-palace_7925318070_o.jpg' (already exists) 
DEBU[2023-06-18T22:37:31Z] metadata: 8045b8926260372bb9afd65e451633ec4ab08817_exiftool.json not found (no such file or directory) 
DEBU[2023-06-18T22:37:31Z] metadata: found no exif header in 20220415_030332_4C1DD8E8.gif.jpg 
INFO[2023-06-18T22:37:31Z] media: millennium-bridge--london_7928216398_o.jpg was taken at 2012-09-03 12:22:44.14 +0000 UTC (meta) 
INFO[2023-06-18T22:37:31Z] import: millennium-bridge--london_7928216398_o.jpg is identical to 2012/09/20120903_122244_BA7B704C.jpg (sha1 187b0d9dd1220fd5a2b14b6f97fa83d969504d36) 
DEBU[2023-06-18T22:37:31Z] media: opened Foo/2022/04/20220415_030332_4C1DD8E8.gif.jpg [360 kB] 
INFO[2023-06-18T22:37:31Z] import: deleted 'Google Takeout 22.01.2001/millennium-bridge--london_7928216398_o.jpg' (already exists) 
DEBU[2023-06-18T22:37:31Z] media: created no new thumbnails for Foo/2022/04/20220415_030332_4C1DD8E8.gif.jpg [38.027019ms] 
INFO[2023-06-18T22:37:31Z] media: 20220415_030332_4C1DD8E8.gif was taken at 2022-04-15 03:03:32 +0000 UTC (file mod time) 
INFO[2023-06-18T22:37:31Z] import: added main gif file Foo/2022/04/20220415_030332_4C1DD8E8.gif 
DEBU[2023-06-18T22:37:31Z] metadata: 8045b8926260372bb9afd65e451633ec4ab08817_exiftool.json not found (no such file or directory) 
DEBU[2023-06-18T22:37:31Z] metadata: found no exif header in 20220415_030332_4C1DD8E8.gif.jpg 
DEBU[2023-06-18T22:37:31Z] media: opened Foo/2022/04/20220415_030332_4C1DD8E8.gif.jpg [360 kB] 
DEBU[2023-06-18T22:37:31Z] media: created no new thumbnails for Foo/2022/04/20220415_030332_4C1DD8E8.gif.jpg [2.55924ms] 
INFO[2023-06-18T22:37:31Z] media: 20220415_030332_4C1DD8E8.gif.jpg was taken at 2023-06-18 22:37:30.982380159 +0000 UTC (file mod time) 
DEBU[2023-06-18T22:37:34Z] photo: Foo/2022/04/20220415_030332_4C1DD8E8 has new title 'Animated Earth' [1.75088ms] 
INFO[2023-06-18T22:37:34Z] import: stacked related jpg file Foo/2022/04/20220415_030332_4C1DD8E8.gif.jpg 
INFO[2023-06-18T22:37:34Z] import: deleted empty folder 'Google Takeout 22.01.2001' 
INFO[2023-06-18T22:37:34Z] import: deleted empty folder 2011/10         
INFO[2023-06-18T22:37:34Z] import: deleted empty folder 2011            
INFO[2023-06-18T22:37:34Z] completed in 7.553374938s                    
```

Note how the files are imported from the *import* folder into the `Foo` subdirectory in the *orginals* folder.

**Reply by lastzero** [reply]
In your case, the command should be somewhat like this instead of what you did:
```
docker exec -ti photoprism photoprism import --dest=Organizar
```

**Reply by ricardopaiva** [reply]
I'm not following you.
I want to import from the default folder (/photoprism/import) to a subfolder in the originals folder (/photoprism/originals/Organize).
Is it possible using the cli?

I also tried these two different options:
docker exec -ti photoprism photoprism --import-dest './Organizar' import
docker exec -ti photoprism photoprism --import-dest './Organizar' import --dest './Organizar'

It just don't work. I always get the same output:

INFO[2023-06-18T22:42:33Z] moving media files from /photoprism/import to **/photoprism/originals/Organizar** => Seems like is going to copy the media files to the correct folder.

But then the output is:
INFO[2023-06-18T22:42:36Z] import: moving main jpg file 994E28F2-CE74-4010-AD9C-EA567C130793.JPG to 2023/06/20230609_120921_A72C0CD8.jpg =>  Is copied to /photoprism/originals/2023/06 instead of **/photoprism/originals/Organizar/2023/06**.

Seems like the import-dest and --dest flags are just ignored when running the "photoprism import" command.

**Comment by ricardopaiva** [comment]
> In your case, the command should be somewhat like this instead of what you did:
> 
> ```
> docker exec -ti photoprism photoprism import --dest=Organizar
> ```

It's not working.
I tried:
`docker exec -ti photoprism photoprism import --dest=Foo`

Output:
```
INFO[2023-06-18T23:07:16Z] config: case-insensitive file system detected 
INFO[2023-06-18T23:07:16Z] Become a member today, support our mission and enjoy our member benefits! 💎 
INFO[2023-06-18T23:07:16Z] Visit https://www.photoprism.app/membership to learn more. 
INFO[2023-06-18T23:07:16Z] migrate: running database migrations         
INFO[2023-06-18T23:07:17Z] moving media files from /photoprism/import to /photoprism/originals/Foo 
INFO[2023-06-18T23:07:17Z] classify: loading nasnet                     
INFO[2023-06-18T23:07:20Z] classify: loading labels from labels.txt     
INFO[2023-06-18T23:07:20Z] import: found no .ppignore file              
INFO[2023-06-18T23:07:20Z] media: 994E28F2-CE74-4010-AD9C-EA567C130793.JPG was taken at 2023-06-09 12:09:21.515 +0000 UTC (meta) 
INFO[2023-06-18T23:07:20Z] import: 994E28F2-CE74-4010-AD9C-EA567C130793.JPG is identical to 2023/06/20230609_120921_A72C0CD8.jpg (sha1 2fb4c56f95cbf37b4a75c1d7b29c52bdfed3331f) 
INFO[2023-06-18T23:07:20Z] import: deleted 994E28F2-CE74-4010-AD9C-EA567C130793.JPG (already exists) 
INFO[2023-06-18T23:08:21Z] completed in 1m4.895833973s     
```

The Foo folder is not created.

I'm running the latest version:
Build [230615-90a18f6e7]

Don't know what's going on here :|

**Reply by ricardopaiva** [reply]
`
docker exec -ti photoprism ls /photoprism/originals -la
`

Output:
```
drwxr-xr-x 1 root       root       4096 Jun  7 07:41 ..
drwxr-xr-x 3 root       root       4096 Jun 18 23:05 2023      -> Newly created folder. Contains the newly imported photos.
drwxrwxrwx 2 root       root          0 Jun 10 11:42 Organizar     -> Previously created. Is empty after import.
drwxrwxrwx 2 root       root          0 Feb 17 18:22 Photos
drwxrwxrwx 2 root       root          0 Dec 14  2022 Videos
```

**Reply by lastzero** [reply]
See the log output. It's because these files already exist in the originals folder. In that case they won't get moved by the import command. However, you can simply move and rename them however you like and then reindex.

**Comment by ricardopaiva** [comment]
> See the log output. It's because these files already exist in the originals folder. In that case they won't get moved by the import command. However, you can simply move and rename them however you like and then reindex.

Man... yes, you're right!
I was deleting the imported files from the originals folder but guess the "problem" was that the files were already indexed so the import command was ignoring the flag (it was importing the file to the same folder where it was before).
I tried with a new image and now it worked fine.

Great news! Thanks so much for your help and sorry for wasting your time. I'm truly happy :)
